### ISSUE TITLE
Summary page doesn't show up

### ISSUE THREAD

**Issue by fabriziocacicia** [guydavis/machinaris#578]
**Describe the bug**
The summary page doesn't show up, it remains stuck at the message:
 _"No blockchains found from any farmers. Just starting up?
Please allow at least 15 minutes for blockchains to get started."_

**To Reproduce**

Not figured out yet.

**Expected behavior**
The summary page should properly load up and show.

**System setup:**
 - OS: 20.04.3 LTS (GNU/Linux 5.13.0-28-generic x86_64)
 - Docker version number: 20.10.12
 - Machinaris branch: main
 - Machinaris version number: 0.6.9

**Config**
<details><summary>full configuration</summary>

```
version: '3.7' 
services: 
    machinaris: 
        image: ghcr.io/guydavis/machinaris:latest 
        container_name: machinaris 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris:/root/.chia 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - worker_address=192.168.1.35 
            - plots_dir=/farming
            - blockchains=chia 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8926:8926 
            - 8927:8927 
            - 8444:8444 
            - 8447:8447 
    
    machinaris-btcgreen: 
        image: ghcr.io/guydavis/machinaris-btcgreen:latest 
        container_name: machinaris-btcgreen 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-btcgreen:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8938 
            - plots_dir=/farming 
            - blockchains=btcgreen 
            - blockchain_db_download=true 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8938:8938 
            - 9282:9282 
            - 18655:18655 
        depends_on: 
            - machinaris 
    
    machinaris-cactus: 
        image: ghcr.io/guydavis/machinaris-cactus:latest 
        container_name: machinaris-cactus 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-cactus:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8936 
            - plots_dir=/farming
            - blockchains=cactus 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8936:8936 
            - 11444:11444 
            - 11447:11447 
        depends_on: 
            - machinaris 
 
    machinaris-cryptodoge: 
        image: ghcr.io/guydavis/machinaris-cryptodoge:latest 
        container_name: machinaris-cryptodoge 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-cryptodoge:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8937 
            - plots_dir=/farming
            - blockchains=cryptodoge 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8937:8937 
            - 15994:15994 
            - 16895:16895 
        depends_on: 
            - machinaris 
    
    machinaris-flax: 
        image: ghcr.io/guydavis/machinaris-flax:latest 
        container_name: machinaris-flax 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-flax:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8928 
            - plots_dir=/farming
            - blockchains=flax 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8928:8928 
            - 6888:6888 
            - 6885:6885 
        depends_on: 
            - machinaris 
            
    
    machinaris-flora: 
        image: ghcr.io/guydavis/machinaris-flora:latest 
        container_name: machinaris-flora 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-flora:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8932 
            - plots_dir=/farming
            - blockchains=flora 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8932:8932 
            - 18644:18644 
            - 18647:18647 
        depends_on: 
            - machinaris 
 
    machinaris-hddcoin: 
        image: ghcr.io/guydavis/machinaris-hddcoin:latest 
        container_name: machinaris-hddcoin 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-hddcoin:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8930 
            - plots_dir=/farming
            - blockchains=hddcoin 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8930:8930 
            - 28444:28444 
            - 28447:28447 
        depends_on: 
            - machinaris 
            
    machinaris-maize: 
        image: ghcr.io/guydavis/machinaris-maize:latest 
        container_name: machinaris-maize 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-maize:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8933 
            - plots_dir=/farming 
            - blockchains=maize 
            - blockchain_db_download=true 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8933:8933 
            - 8644:8644 
            - 8647:8647 
        depends_on: 
            - machinaris   
            
    machinaris-shibgreen: 
        image: ghcr.io/guydavis/machinaris-shibgreen:latest 
        container_name: machinaris-shibgreen 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-shibgreen:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8939 
            - plots_dir=/farming 
            - blockchains=shibgreen 
            - farmr_skip_launch=true
            - bladebit_skip_build=true 
        ports: 
            - 8939:8939 
            - 7442:7442 
            - 18974:18974 
        depends_on: 
            - machinaris       
             
    machinaris-staicoin: 
        image: ghcr.io/guydavis/machinaris-staicoin:latest 
        container_name: machinaris-staicoin 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-staicoin:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8934 
            - plots_dir=/farming
            - blockchains=staicoin 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8934:8934 
            - 1999:1999 
            - 1692:1692 
        depends_on: 
            - machinaris 
    machinaris-stor: 
        image: ghcr.io/guydavis/machinaris-stor:latest 
        container_name: machinaris-stor 
        hostname: pietrecadute 
        restart: always 
        volumes: 
            - ~/.machinaris-stor:/root/.chia 
            - ~/.machinaris/mnemonic.txt:/root/.chia/mnemonic.txt:ro 
            - "/mnt/ssd:/plotting/ssd" 
            - "/farming:/farming"
        environment: 
            - TZ=Europe/Rome 
            - mode=fullnode 
            - controller_host=192.168.1.35 
            - worker_address=192.168.1.35 
            - worker_api_port=8935 
            - plots_dir=/farming
            - blockchains=stor 
            - farmr_skip_launch=true
            - bladebit_skip_build=true
        ports: 
            - 8935:8935 
            - 8668:8668 
            - 8337:8337 
        depends_on: 
            - machinaris  
```

</details>

**Additional context & screenshots**
I'm attaching the logs from boot to the issue.
[apisrv.log](https://github.com/guydavis/machinaris/files/8072160/apisrv.log)
[webui.log](https://github.com/guydavis/machinaris/files/8072161/webui.log)

**Comment by guydavis** [comment]
Good day!  Thanks for the detailed report.  I can see in the logs, a Python error due to missing data.  

```
2022-02-15 15:21:11 [84] [ERROR] Error loading summary: 'NoneType' object is not subscriptable
```

Unfortunately, I didn't do a good job of logging the exact cause so it's not clear what the root cause is.  

I have just updated the `:develop` image with improved logging in this section.  If you wouldn't mind, please try the [development stream](https://github.com/guydavis/machinaris/wiki/Releases#release-streams).  After upgrading, open the Web Log on the Workers page (Chia fullnode row, right-side), then click the Summary link and check the log.  This should tell me the fix or guard I need to put into the Summary data pull.  Thanks!

**Comment by guydavis** [comment]
Thanks to Discord user "WibbleWibble", I have enhanced the Summary page table to guard for data lookup in each and every cell of the table.  This means the Summary table will now show, but some cells will be empty.  Much better than the entire table being missing because of some bad status.

Most importantly, the Web Log (Workers page - Chia fullnode row) now will log exactly which Summary table cell is causing a problem.  In the case of user:WibbleWibble, he found his Chives blockchain wallet was not synching correctly.  He was able to fix the wallet problem and the missing data on the Summary page appeared.

This improved guard/logging on the Summary page is available in the `:develop` image now.  Will be in `:test` next week, then in the upcoming release.  Thanks all!

**Comment by guydavis** [comment]
Closing as now generally available in the [Machinaris v0.7.0](https://github.com/guydavis/machinaris/releases/tag/0.7.0) release.  Thanks!
